### PR TITLE
User activation. r=fabrice

### PR DESCRIPTION
**Pull request by ferjm**

### CODE REVIEW

**Comment by arcturus** [comment]
LGTM r+!

**Comment by ferjm** [comment]
It's not ready yet :D

**Comment by arcturus** [comment]
Lol, I thought it was just a renaming of the auth endpoint :) ... so this will have the sending email and so on right?

**Comment by ferjm** [comment]
Yes, this will include sending the invitation email.

**Comment by ferjm** [comment]
Now it's ready :) r? @arcturus

**Comment by ferjm** [comment]
This patch allows the consumer of the crate to provide a callback that will receive a user email and an activation url for each new user registration. For the FoxBox's case, we will be sending invitation emails to these email address and with the activation urls as bodies of these emails.

**Comment by ferjm** [comment]
I ended up choosing a simpler API where the consumer only needs to provide the email server url and the invitation url prepath. This way the users crate is responsible for the actual send of the invitation emails.

File: src/invitation_middleware.rs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                return;
+            }
+        };
+
+        let client = Client::new();
+        let endpoint = format!("{}/v1/invitation", email_server);
+        let mut headers = Headers::new();
+        headers.set(
+            ContentType(Mime(TopLevel::Application, SubLevel::Json,
+                             vec![(Attr::Charset, Value::Utf8)]))
```
Comment (fabricedesre) [inline src/invitation_middleware.rs]: Switch to http://hyper.rs/hyper/v0.9.8/hyper/header/struct.ContentType.html#method.json instead

File: src/invitation_middleware.rs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+#[derive(Clone, Debug)]
+pub struct InvitationMiddleware{
+    email_server: Option<String>,
+    invitation_url_prepath: Option<String>,
+    version: String
+}
+
+impl InvitationMiddleware {
+    pub fn new(version: String) -> Self {
```
Comment (fabricedesre) [inline src/invitation_middleware.rs]: Here and in other methods, use &str as a parameter instead of String

**Pull request merged**